src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -74,6 +75,13 @@ internal sealed partial class TagComputer
 
         private int _taggerReferenceCount;
 
+        /// <summary>
+        /// Cached values for the last services we computed for a particular <see cref="Workspace"/> and <see
+        /// cref="IContentType"/>.  These rarely change, and are expensive enough to show up in very hot scenarios (like
+        /// scrolling) where we are going to be called in at a very high volume.
+        /// </summary>
+        private Tuple<Workspace, IContentType, (SolutionServices solutionServices, IClassificationService classificationService)>? _lastCachedServices;
```

Comment:
💡 A `record` would be preferred here, since it means the properties have better names and are read-only.

Reply:
done.